File: FurnitureStore/FurnitureStore.Models/Subcategory/Subcategory.cs

namespace FurnitureStore.Models.Subcategory
{
    public class Subcategory
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public long CategoryId { get; set; }
        public Category.Category Category { get; set; }

    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/subcategory/subcategory.dart
import 'package:json_annotation/json_annotation.dart';

part 'subcategory.g.dart';

@JsonSerializable()
class SubcategoryModel {
  int? id;
  String? name;
  int? categoryId;
  String? imagePath;

  SubcategoryModel(this.id, this.name, this.categoryId, this.imagePath);

  factory SubcategoryModel.fromJson(Map<String, dynamic> json) =>
      _$SubcategoryModelFromJson(json);

  Map<String, dynamic> toJson() => _$SubcategoryModelToJson(this);
}
